### ISSUE TITLE
Add filters to ec2 describe-instances

### ISSUE THREAD

**Issue by srakhe-uptycs** [localstack/localstack#2646]
# Type of request: This is a ...

[x] bug report

# Detailed description

Localstack does not return expected output when using filters on describe-instances in ec2.
...

## Expected behavior

Command:
`aws --endpoint-url=http://localhost:4566 ec2 describe-instances --filters owner-id=111111111111`

Output:

`{
    "Reservations": [
        {
            "Groups": [],
            "Instances": [
                {
                    "AmiLaunchIndex": 0,
                    "ImageId": "ami-00dc79254d1111110",
                    "InstanceId": "i-00d11111111111a80",
                    "InstanceType": "t2.micro",
                    "KeyName": "newdemo",
                    "LaunchTime": "2020-04-29T20:58:13+00:00",
                    "Monitoring": {
                        "State": "disabled"
                    },
                    "Placement": {
                        "AvailabilityZone": "us-east-1",
                        "GroupName": "",
                        "Tenancy": "default"
                    },
                    "PrivateDnsName": "ip-11-11-0-11.ec2.internal",
                    "PrivateIpAddress": "11.1.1.11",
                    "ProductCodes": [],
                    "PublicDnsName": "ec2-12-12-12-12.com",
                    "PublicIpAddress": "12.12.12.12",
                    "State": {
                        "Code": 16,
                        "Name": "running"
                    },
                    "StateTransitionReason": "",
                    "SubnetId": "subnet-123123123123",
                    "VpcId": "vpc-0db1111111dfcd698",
                    "Architecture": "x86_64",
                    "BlockDeviceMappings": [
                        {
                            "DeviceName": "/dev/xvda",
                            "Ebs": {
                                "AttachTime": "2020-04-29T20:58:14+00:00",
                                "DeleteOnTermination": true,
                                "Status": "attached",
                                "VolumeId": "vol-011111111111db"
                            }
                        },
                        {
                            "DeviceName": "/dev/sdh",
                            "Ebs": {
                                "AttachTime": "2020-04-29T20:58:33+00:00",
                                "DeleteOnTermination": false,
                                "Status": "attached",
                                "VolumeId": "vol-01550111111111"
                            }
                        }
                    ],
                    "ClientToken": "newde-Asset-SVLAMZZZZIS8",
                    "EbsOptimized": false,
                    "EnaSupport": true,
                    "Hypervisor": "xen",
                    "IamInstanceProfile": {
                        "Arn": "arn:aws:iam::111111111111:instance-profile/newdemo-EC2InstanceProfile-111O011110315",
                        "Id": "AAAA5SBBSGA7EQJFSPQUK"
                    },
                    "NetworkInterfaces": [
                        {
                            "Association": {
                                "IpOwnerId": "amazon",
                                "PublicDnsName": "ec2-12-12-12-12.compute-1.amazonaws.com",
                                "PublicIp": "12.12.12.12"
                            },
                            "Attachment": {
                                "AttachTime": "2020-04-29T20:58:13+00:00",
                                "AttachmentId": "eni-attach-073515afbc6904407",
                                "DeleteOnTermination": true,
                                "DeviceIndex": 0,
                                "Status": "attached"
                            },
                            "Description": "",
                            "Groups": [
                                {
                                    "GroupName": "WebSg",
                                    "GroupId": "sg-062f17a512d3a9d6f"
                                }
                            ],
                            "Ipv6Addresses": [],
                            "MacAddress": "0e:cf:ae:23:9b:e1",
                            "NetworkInterfaceId": "eni-0c65394d55862590c",
                            "OwnerId": "921341669438",
                            "PrivateDnsName": "ip-17-1-0-14.ec2.internal",
                            "PrivateIpAddress": "17.1.0.14",
                            "PrivateIpAddresses": [
                                {
                                    "Association": {
                                        "IpOwnerId": "amazon",
                                        "PublicDnsName": "ec2-12-12-12-12.compute-1.amazonaws.com",
                                        "PublicIp": "12.12.12.12"
                                    },
                                    "Primary": true,
                                    "PrivateDnsName": "ip-17-1-0-14.ec2.internal",
                                    "PrivateIpAddress": "17.1.0.14"
                                }
                            ],
                            "SourceDestCheck": true,
                            "Status": "in-use",
                            "SubnetId": "subnet-0024788cf4e41d62d",
                            "VpcId": "vpc-0db5918790dfcd698",
                            "InterfaceType": "interface"
                        }
                    ],
                    "RootDeviceName": "/dev/xvda",
                    "RootDeviceType": "ebs",
                    "SecurityGroups": [
                        {
                            "GroupName": "WebSg",
                            "GroupId": "sg-062f17a512d3a9d6f"
                        }
                    ],
                    "SourceDestCheck": true,
                    "Tags": [
                        {
                            "Key": "Name",
                            "Value": "newdemo-us-east-1-asset05"
                        },
                        {
                            "Key": "aws:cloudformation:stack-name",
                            "Value": "newdemo"
                        },
                        {
                            "Key": "aws:cloudformation:logical-id",
                            "Value": "Asset05"
                        },
                        {
                            "Key": "aws:cloudformation:stack-id",
                            "Value": "arn:aws:cloudformation:us-east-1:111111111111:stack/newdemo/c98a1bf0-8a5b-11ea-a486-0ee1cf96ccc3"
                        }
                    ],
                    "VirtualizationType": "hvm",
                    "CpuOptions": {
                        "CoreCount": 1,
                        "ThreadsPerCore": 1
                    },
                    "CapacityReservationSpecification": {
                        "CapacityReservationPreference": "open"
                    },
                    "HibernationOptions": {
                        "Configured": false
                    },
                    "MetadataOptions": {
                        "State": "applied",
                        "HttpTokens": "optional",
                        "HttpPutResponseHopLimit": 1,
                        "HttpEndpoint": "enabled"
                    }
                }
            ],
            "OwnerId": "111111111111",
            "RequesterId": "222222222222",
            "ReservationId": "r-133354efc510b434t"
        }
    ]
}`

...

## Actual behavior

Command used:
`aws --endpoint-url=http://localhost:4566 ec2 describe-instances --filters Name=owner-id Values=000000000000`
(owner-id is 000000000000 by default in ec2 instance for localstack)

Output:
```
Unable to parse response (syntax error: line 1, column 54), invalid XML received. Further retries may succeed:
b'<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">\n<title>500 Internal Server Error</title>\n<h1>Internal Server Error</h1>\n<p>The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.</p>\n'
```

...

# Steps to reproduce

## Command used to start LocalStack

Started localstack locally with `make infra` and also tried with docker: `sudo docker-compose up`

## Instance creation and verification:

Instance created with `run-instances` and the creation was verified using `describe-instances`

...

**Comment by srakhe-uptycs** [comment]
@whummer I want to know if this is possible through modification of moto code for ec2. And if so I'd like to contribute to this. Thanks.

**Comment by whummer** [comment]
Thanks for reporting @srakhe-uptycs . Yes, I believe this could be fixed on moto side - would be great if you could contribute a pull request. 👍  Thanks!

**Comment by whummer** [comment]
@srakhe-uptycs This has been fixed upstream in spulec/moto#3149 - can you please give it another try with the `latest` version of the Docker image? Please report here if the problem persists. Thanks
